cannon/mipsevm/exec/mips_instructions64_test.go

```diff
@@ -109,3 +109,38 @@ func TestStoreSubWord_64bits(t *testing.T) {
 		})
 	}
 }
+
+func TestSignExtend_64bit(t *testing.T) {
+	cases := []struct {
+		name     string
+		data     Word
+		index    Word
+		expected Word
+	}{
+		{name: "idx 32, signed", data: 0x0000_0000_8123_4567, index: 32, expected: 0xFFFF_FFFF_8123_4567},
+		{name: "idx 32, unsigned", data: 0x0000_0000_7123_4567, index: 32, expected: 0x0000_0000_7123_4567},
+		{name: "idx 32, signed, non-zero upper bits", data: 0x1234_4321_8123_4567, index: 32, expected: 0xFFFF_FFFF_8123_4567},
+		{name: "idx 32, unsigned, non-zero upper bits", data: 0xFFFF_FFFF_7123_4567, index: 32, expected: 0x0000_0000_7123_4567},
```

Comment:
Sign-extension doing zero-extension is unintuitive and a potential footgun. However, there aren't any callers of `SignExtend` today that use the upper word (bits 63..32). So this is OK for now.
It's too late to do this for the code freeze, but later we should consider updating `SignExtend` to leave the upper bits untouched if no sign-extension is needed.

Reply:
i was mistaken, zero extension is used if the bit is empty per the spec. thanks for clarifying this offline.

Reply:
Just double-checked the MIPS spec.  It looks like the `sign_extend` function will zero- or sign-extend values:
<img width="502" alt="Screenshot 2025-02-27 at 10 40 02 AM" src="https://github.com/user-attachments/assets/e437c644-71c1-4de9-ac6b-2bc53bd1882e" />
^ grabs `val[nbits - 1]` and repeats